mlflow/tensorflow/__init__.py

```diff
@@ -1046,25 +1046,35 @@ def _patch_implementation(
             unlogged_params = ["self", "x", "y", "callbacks", "validation_data", "verbose"]
 
             batch_size = None
-            training_data = kwargs["x"] if "x" in kwargs else args[0]
-            if isinstance(training_data, tensorflow.data.Dataset):
-                batch_size = training_data._batch_size.numpy()
-            elif isinstance(training_data, tensorflow.keras.utils.Sequence):
-                first_batch_inputs, _ = training_data[0]
-                batch_size = len(first_batch_inputs)
-            elif is_iterator(training_data):
-                peek = next(training_data)
-                batch_size = len(peek[0])
-
-                def __restore_generator(prev_generator):
-                    yield peek
-                    yield from prev_generator
-
-                restored_generator = __restore_generator(training_data)
-                if "x" in kwargs:
-                    kwargs["x"] = restored_generator
-                else:
-                    args = (restored_generator,) + args[1:]
+            try:
+                training_data = kwargs["x"] if "x" in kwargs else args[0]
+                if isinstance(training_data, tensorflow.data.Dataset) and hasattr(
+                    training_data, "_batch_size"
+                ):
+                    batch_size = training_data._batch_size.numpy()
+                elif isinstance(training_data, tensorflow.keras.utils.Sequence):
+                    first_batch_inputs, _ = training_data[0]
+                    batch_size = len(first_batch_inputs)
```

Comment:
Did we want to log the `drop_remainder` param for `N % batch_size` non-evenly divisible data set length or is that just superfluous information for autologging batch information?

Reply:
Good question! IMO, it's superfluous for autologging batch info. If we were to add it, I'd prefer to do it via a new PR in the future to avoid future regressions related to extracting attributes of `tensorflow.data.Dataset` :).

Reply:
yep definitely not in this one. Included the comment for posterity sake only ;)